python/ray/train/checkpoint.py

```diff
@@ -186,7 +186,19 @@ def as_directory(self) -> Iterator[str]:
 
         Example:
 
-        .. code-block:: python
+        .. testcode::
+            :hide:
+
+            from pathlib import Path
+            import tempfile
+
+            from ray.train.checkpoint import Checkpoint
```

Comment:
Please don't do this, see https://github.com/ray-project/ray/pull/37925#issuecomment-1661163620

Reply:
I'm currently forced to do this, since 2 "Checkpoint" classes exist at the moment:

```
from ray.air import Checkpoint
from ray.train._checkpoint import Checkpoint

__all__ = ["Checkpoint"]  # These 2 checkpoints are colliding -- I can't expose both the same way.
```

By 2.7, I will definitely make the switch for `ray.train.Checkpoint` to point to `ray.train._checkpoint.Checkpoint`.
But, we're not 100% ready to make the switch just yet (it will involve a bunch of tests to change, and the new Checkpoint is not fully supported yet by other internal components).

For now, I can add a TODO in this docstring to update it to `ray.train.Checkpoint` when the switch has happened. How does that sound?

Reply:
I'm also happy to pre-emptively rename this to `_checkpoint.py` so that the `ray.train.checkpoint` namespace is no longer a thing.

Reply:
Sounds good, why don't you rename this to `_checkpoint` while this is still brewing to avoid confusion? :)

Reply:
Done! PTAL